### PR TITLE
Fix ubuntu trigger paths to not use wildcard for version folder name

### PR DESCRIPTION
**Pull request by mthalman**
The build triggers aren't working for Ubuntu, apparently because a wildcard is used in the folder name.  Changing it to be explicit for the version name instead.  This is technically the right thing to do anyway in case we were to ever add support for additional versions with the same major version, in which case we'd want separate pipelines for each.

### CODE REVIEW

File: .vsts-pipelines/dotnet-buildtools-prereqs-ubuntu14.yml [hunk 1]
```diff
@@ -4,14 +4,14 @@ trigger:
     - master
   paths:
     include:
-    - src/ubuntu/14*
+    - src/ubuntu/14.04/*
 pr:
```
Comment (MichaelSimons) [inline .vsts-pipelines/dotnet-buildtools-prereqs-ubuntu14.yml]: This comment is orthogonal to these changes.  Should these pr triggers be removed?  They are a little mis-leading.

Comment (mthalman) [reply .vsts-pipelines/dotnet-buildtools-prereqs-ubuntu14.yml]: I could go either way.  They still can apply if we were to ever do a pull request on the internal repo.

Comment (MichaelSimons) [reply .vsts-pipelines/dotnet-buildtools-prereqs-ubuntu14.yml]: My preference is to clean up dead/unused code paths as they are a maintenance tax and add complexity to the code base.  In this particular case, it adds confusion as to why this is needed in addition to https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/master/.vsts-pipelines/dotnet-buildtools-prereqs-pr.yml.  If needed in the future it would be easy to add back.
